src/GafferArnold/IECoreArnoldPreview/Renderer.cpp

```diff
@@ -1378,23 +1379,18 @@ class ArnoldObject : public IECoreScenePreview::Renderer::ObjectInterface
 		void applyTransform( AtNode *node, const std::vector<Imath::M44f> &samples, const std::vector<float> &times )
 		{
 			const size_t numSamples = samples.size();
-			AtArray *timesArray = AiArrayAllocate( samples.size(), 1, AI_TYPE_FLOAT );
 			AtArray *matricesArray = AiArrayAllocate( 1, numSamples, AI_TYPE_MATRIX );
 			for( size_t i = 0; i < numSamples; ++i )
 			{
-				AiArraySetFlt( timesArray, i, times[i] );
 				AiArraySetMtx( matricesArray, i, *reinterpret_cast<AtMatrix*>(
 					const_cast<float (*)[4][4]>( &samples[i].x ) ) );
 			}
 			AiNodeSetArray( node, "matrix", matricesArray );
-			if( AiNodeEntryLookUpParameter( AiNodeGetNodeEntry( node ), "transform_time_samples" ) )
-			{
-				AiNodeSetArray( node, "transform_time_samples", timesArray );
-			}
-			else
-			{
-				AiNodeSetArray( node, "time_samples", timesArray );
-			}
+
+			NodeAlgo::ensureUniformTimeSamples( times );
```

Comment:
Hmm, if you're generating scenes using Gaffer, it should be impossible to generate that case, right?  In fact, it should be impossible to generate any case that fails ensureUniformSampleTimes either, using Gaffer.

But I guess since we still theoretically support standalone Cortex I should see if there's any way I could flag this without too much difficulty.

Reply:
Yeah, Gaffer should be fine, but in theory anyone could use the renderer backends - one outcome of the Encapsulate work is that we will be able to support Python procedurals in Cortex 10 for any die hards that want them (albeit with the new API rather than the old one).

You know what though, I don't see a way of implementing this warning without storing additional state, so unless you can see a way of doing it without overhead, let's not worry about it.